File: semestr1/Lab06/Python/main.py
import permutacjelib
import sys



def main():
    if(len(sys.argv) != 2):
        print("Zła liczba argumentów")
        return
    
    n = int(sys.argv[1])

    if(n <= 1):
        print("Błędny argument")
        return

    permutacje = [i for i in range(1, n+1)]

    solutions = 0

    wynik = permutacjelib.isSolution(permutacje)

    if(wynik):
        solutions += 1
        for i in permutacje:
            print(f"{i} ", end="")
    
    while (permutacjelib.nextPerm(permutacje)):
        wynik = permutacjelib.isSolution(permutacje)

        if(wynik):
            solutions += 1
            for i in permutacje:
                print(f"{i} ", end="")
            print("")
    

    print(f"Number of solutions: {solutions}")


if __name__ == "__main__":
    main()

File: semestr1/Lab06/C/main.c
#include <stdio.h>
#include <stdlib.h>

#include "permutacje.h"



int main(int argc, char* argv[]){
    
    if (argc != 2){
        printf("Zła liczba argumentów\n");
        return 1;
    }

    int n = atoi(argv[1]);
    
    if (n <= 1){
        printf("Błędny argument\n");
        return 1;
    }

    int permutacje[n];
    
    genTab(permutacje, n);
    
    int solutions = 0;
    bool wynik = isSolution(permutacje, n);

    if (wynik){
        solutions = solutions + 1;
        for(int i = 0; i < n; i++){
            printf("%d", permutacje[i]);
        }
    }

    while (nextPerm(permutacje, n)){
        wynik = isSolution(permutacje, n);
        if (wynik){
            solutions = solutions + 1;
            for(int i = 0; i < n; i++){
                printf("%d ", permutacje[i]);
            }
            printf("\n");
        }        
    }
    printf("Number of solutions: %d\n", solutions);
}